src/renderer/components/getDi.tsx

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2021 OpenLens Authors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { createContainer } from "@ogre-tools/injectable";
+import type { IConfigurableDependencyInjectionContainer } from "@ogre-tools/injectable";
+
+export const getDi = () => {
```

Comment:
Can we see an example of how would typical react component would look with all that stuff?
And maybe not only react components, all known use-cases would be good to see. Thanks!

Reply:
Certainly! Working on it :)

Reply:
New component from scratch: https://github.com/jansav/lens/commit/e8fca8d5a4df30f7c0151c54acf743ecd6033b51
Apologies for lack of imagination. It's another TODO-list.

Reply:
For more, check out detailed examples for every feature here: 
- [injectable-react library](https://github.com/ogre-works/ogre-tools/blob/master/packages/injectable-react/src/Inject/Inject.test.js)
- [injectable library](https://github.com/ogre-works/ogre-tools/blob/master/packages/injectable/src/dependency-injection-container/createContainer.test.js)

Reply:
It's better if the examples using `react-testing-library` instead of `enzyme`, at least I don't understand what `enzymeUpdate` do in the examples.